modules/ximgproc/src/fourier_descriptors.cpp

```diff
@@ -0,0 +1,334 @@
+{
+    CV_Assert(_src.kind() == _InputArray::STD_VECTOR || _src.kind() == _InputArray::MAT);
+    CV_Assert(_src.empty() || (_src.channels() == 2 && (_src.depth() == CV_32S || _src.depth() == CV_32F || _src.depth() == CV_64F)));
+    CV_Assert(nbElt>0);
+    Mat ctr;
+    _src.getMat().convertTo(ctr,CV_32F);
+    if (ctr.rows*ctr.cols == 0)
+    {
+        _out.release();
+        return;
+    }
+    CV_Assert(ctr.rows==1 || ctr.cols==1);
+    double		l1 = 0, l2, p, d, s;
+ //   AutoBuffer<Point2d> _buf(nbElt);
+    Mat r;
+    if (ctr.rows==1)
+        ctr=ctr.t();
+    int j = 0;
+    int 		nb = ctr.rows;
+    p = arcLength(_src, true);
+    l2 = norm(ctr.row(j) - ctr.row(j + 1)) / p;
+    for (int i = 0; i<nbElt; i++)
+    {
+        s = (float)i / (float)nbElt;
+        while (s >= l2)
+        {
+            j++;
+            l1 = l2;
+            d = norm(ctr.row(j % nb) - ctr.row((j + 1) % nb));
+            l2 = l1 + d / p;
+        }
+        if ((s >= l1) && (s < l2))
+        {
+            Mat d1=ctr.row((j + 1) % nb);
+            Mat d0=ctr.row(j % nb);
+            Mat d10 = d1 - d0;
+            Mat pn = d0 + d10 * (s - l1) / (l2 - l1);
+            r.push_back(pn);
+ //           _buf[i]=Point2d(pn.at<Point2f>(0,0));
+        }
+    }
+    r.copyTo(_out);
+}
+
+void transform(InputArray _src, InputArray _t,OutputArray _dst,  bool fdContour)
+{
+    if (!fdContour)
+        CV_Assert(_src.kind() == _InputArray::STD_VECTOR);
+    else
+        CV_Assert( _src.kind() == _InputArray::MAT );
+    CV_Assert(_src.channels() == 2);
+    CV_Assert(_t.kind() == _InputArray::MAT);
+    Mat t=_t.getMat();
+    CV_Assert(t.rows == 1 && t.cols==5 && t.depth()==CV_64F);
+    Mat Z;
+    if (!fdContour)
+    {
+        Mat ctr1 = _src.getMat();
+        if (ctr1.rows==1)
+            ctr1 = ctr1.t();
+        Mat newCtr1;
+        int M = getOptimalDFTSize(ctr1.rows);
+        contourSampling(ctr1, newCtr1, M);
+        fourierDescriptor(newCtr1, Z);
+    }
+    else
+        Z = _src.getMat();
+    if (Z.type()!=CV_64FC2)
+        Z.convertTo(Z,CV_64F);
+    std::complex<double> expitheta = t.at<double>(0,2)  * std::complex<double>(cos(t.at<double>(0, 1)), sin(t.at<double>(0, 1)));
+    for (int j = 1; j<Z.rows; j++)
+    {
+        std::complex<double> zr(Z.at<Vec2d>(j, 0)[0], Z.at<Vec2d>(j,0)[1]);
+        if (j<=Z.rows/2)
+            zr = zr*expitheta*exp(t.at<double>(0, 0) * 2 * (M_PI*j) *  std::complex<double>(0, 1));
+        else
+            zr = zr*expitheta*exp(t.at<double>(0, 0)* 2 * (M_PI*(j - Z.rows))    * std::complex<double>(0, 1));
+        Z.at<Vec2d>(j, 0) = Vec2d(zr.real(),zr.imag());
+    }
+    Z.at<Vec2d>(0, 0) += Vec2d(t.at<double>(0, 3), t.at<double>(0, 4));
+    std::vector<Point2d> z;
+    dft(Z, z, DFT_INVERSE);
+    std::vector<Point> c;
+    for (int j = 0; j<static_cast<int>(z.size()); j++)
+        c.push_back(Point2d(z[j].x, z[j].y));
+    Mat(z).copyTo(_dst);
```

Comment:
Look like "c" is not used here.

> static_cast<int>(z.size())

Static code analyses dislike this (out of range check is expected) - consider to use `size_t` iterator type.